bundle.js

```diff
@@ -126,12 +126,40 @@ const addRequest = (event, bundle, convertedBundle) => {
   const params = _.get(bundle, 'request.params', {});
   _.extend(convertedBundle.request.params, params);
 
-  let body = _.get(bundle, 'request.body');
+  const body = _.get(bundle, 'request.body');
   if (body) {
-    if (typeof body !== 'string') {
-      body = JSON.stringify(body);
+    let data = body,
+      files;
+
+    if (typeof data !== 'string') {
+      if (bundle._fileFieldKeys) {
+        // Exclude file fields from request.data
+        data = Object.keys(body)
+          .filter(k => bundle._fileFieldKeys.indexOf(k) === -1)
+          .reduce((result, k) => {
+            result[k] = body[k];
+            return result;
+          }, {});
+
+        files = Object.keys(body)
+          .filter(k => bundle._fileFieldKeys.indexOf(k) >= 0)
+          .reduce((result, k) => {
+            // TODO: Any way we can tell filename and mimetype from a hydrate URL?
```

Comment:
> That was an acceptable solution on Python side because, after making the request, the response persists in memory so long as the LazyFile object continued to exist.

Does it mean the entire file content is kept in memory? If we do the same in node.js side, wouldn't it exceed the memory limit too easily?

Reply:
It depends. If the API providing the file streams it, then we may only read the headers, and keep the response object so we can continue calling `.read()` to finish streaming it from the server. Some cases though, yes, it will bring the whole file in. And yes, the memory limits on lambda are a concern. I'm not sure how to get around it beyond bumping them up though.